org.lflang.ui/src/org/lflang/ui/LFUiModuleImpl.java

```diff
@@ -0,0 +1,317 @@
+                            newC.text += "    ";
+                            newC.cursorOffset += 4;
+                        }
+                        newC.text += string;
+                        newC.text += command.text.trim();
+                        newC.text += "\n";
+                        for (int i = 0; i < (indentation / 4); i++) {
+                            newC.text += "    ";
+                        }
+                        newC.length += string.length();
+                    } else {
+                        // Creating a new first line within a pre-existing block.
+                        final int indentation = indentationAt(document, command.offset);
+                        int length = 0;
+                        for (int i = 0; i < ((indentation / 4) + 1); i++) {
+                            newC.text += "    ";
+                            newC.cursorOffset += 4;
+                            length += 4;
+                        }
+                        // The length field is, as usual for xtext, undocumented.
+                        // It is not the length of the new command, but seems to be
+                        // the number of characters of the original document that are
+                        // to be replaced.
+                        newC.length = 0;
+                    }
+                }
+                return newC;
+            }
+
+            // Expose base class protected methods within this package.
+            @Override
+            public IRegion findStopTerminal(final IDocument document, final int offset) throws BadLocationException {
+                return super.findStopTerminal(document, offset);
+            }
+
+            @Override
+            public IRegion findStartTerminal(final IDocument document, final int offset) throws BadLocationException {
+                return super.findStartTerminal(document, offset);
+            }
+
+            @Override
+            public void internalCustomizeDocumentCommand(final IDocument document, final DocumentCommand command) {
+                try {
+                    super.internalCustomizeDocumentCommand(document, command);
+                } catch (Throwable e) {
+                    throw Exceptions.sneakyThrow(e);
+                }
+            }
+        }
+
+        /**
+         * Strategy for handling combinations of nested braces of different types.
+         * This is based on CompoundMultiLineTerminalsEditStrategy by Sebastian Zarnekow,
+         * but unfortunately, that class is not written in a way that can be overridden,
+         * so this is a reimplementation.
+         */
+        public static class CompoundLFMultiLineTerminalsEditStrategy extends AbstractTerminalsEditStrategy {
+            /**
+             * Strategies used to handle combinations of nested braces.
+             */
+            private List<LFMultiLineTerminalsEditStrategy> strategies = new LinkedList<LFMultiLineTerminalsEditStrategy>();
```

Comment:
Perhaps this could be a static final list initialized using `List.of`, and perhaps it could be moved so that it is accessible on line 265, "configureMultilineCodeBlock"?

Reply:
Thanks, Peter. I believe I addressed all of your comments including this. Please let me know if there's anything missing.

Reply:
Thanks!